dataactbroker/handlers/account_handler.py

```diff
@@ -470,3 +470,57 @@ def list_user_emails():
 
     user_info = [{"id": user.user_id, "name": user.name, "email": user.email} for user in users]
     return JsonResponse.create(StatusCode.OK, {"users": user_info})
+
+
+def list_submission_users(d2_submission):
+    """ List user IDs and names that have submissions that the requesting user can see.
+
+        Arguments:
+            d2_submission: boolean indicating whether it is a DABS or FABS submission (True if FABS)
+
+        Returns:
+            A JsonResponse containing a list of users that have submissions that the requesting user can see
+    """
+
+    # print(sess.query(Submission).filter(Submission.user_id == 2).exists())
+    sess = GlobalDB.db().session
+    # subquery to create the EXISTS portion of the query
+    exists_query = sess.query(Submission).filter(Submission.user_id == User.user_id,
+                                                 Submission.d2_submission.is_(d2_submission))
+
+    # if user is not an admin, we have to adjust the exists query to limit submissions
+    if not g.user.website_admin:
+        # split affiliations into frec and cgac
+        cgac_affiliations = [aff for aff in g.user.affiliations if aff.cgac]
```

Comment:
Stupid question: if `d2_submission` is true and we're requesting FABS submissions, do we want to filter these with the following:

`cgac_affiliations = [aff for aff in g.user.affiliations if FABS_PERMISSION_ID_LIST]`

I noticed that the FABS_PERMISSION_ID_LIST only works with `submitter`, `editfabs`, and `fabs` whereas `DABS_PERMISSION_ID_LIST` only works with `reader` and `writer`.

Reply:
DABS users can view FABS submissions for their agencies but not the other way around (FABS users can't view DABS submissions without DABS permissions)